src/lib.rs

```diff
@@ -78,6 +92,16 @@ impl UnixStream {
             }
         }
     }
+
+    /// Create an unnamed pair of connected sockets.
+    ///
+    /// Returns two `UnixStream`s which are connected to each other.
+    pub fn unnamed() -> io::Result<[UnixStream; 2]> {
```

Comment:
Do you think it'd make more sense to return a `(UnixStream, UnixStream)` here or a `[UnixStream; 2]`? I could be convinced either way, though a tuple seems a bit more common in APIs.